subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -56,6 +56,15 @@
 
 #define DISCOVER_PREFACE	"</.well-known/core>;ct=40"
 
+#define RESOURCE_TYPE		";rt=\"oma.lwm2m\""
```

Comment:
And add a TODO: to implement a way for clients to specify alternate path via Kconfig